src/BotPlutusInterface/CardanoCLI.hs

```diff
@@ -187,89 +173,33 @@ calculateMinFee pabConf tx =
-  Member (PABEffect w) effs =>
-  [TxId] ->
-  Eff effs (Either Text (Map TxOutRef TxOut))
-queryTxOuts txIds = do
-  res <- queryChainIndex @w $ TxsFromTxIds txIds
-  return $ case res of
-    TxIdsResponse chainTxs -> Right $ foldMap (fmap (sortTxOut . fst) . txOutRefMap) chainTxs
-    _ -> Left "Wrong PAB response"
-  where
-    -- Keep values in TxOuts sorted for deterministic ToData conversion
-    sortTxOut :: TxOut -> TxOut
-    sortTxOut txOut = txOut {txOutValue = sortValue $ txOutValue txOut}
-
--- There is no match txOutRefs request, and we don't want a separate PAB query per input.
--- So, for efficiency, we're going to query the transactions for all inputs combined,
--- then pick out the outputs we care about
-getTxInInfos ::
-  forall (w :: Type) (effs :: [Type -> Type]).
-  Member (PABEffect w) effs =>
-  [TxOutRef] ->
-  Eff effs (Either Text [TxInInfo])
-getTxInInfos txOutRefs = do
-  let ids = nub $ txOutRefId <$> txOutRefs
-  eAllOutRefs <- queryTxOuts @w ids
-  return $
-    eAllOutRefs >>= \allOutRefs ->
-      sequence $ (\ref -> toEither $ TxInInfo ref <$> Map.lookup ref allOutRefs) <$> txOutRefs
-  where
-    toEither :: Maybe TxInInfo -> Either Text TxInInfo
-    toEither = maybeToRight "Couldn't find TxOut"
-
-{- | Builds the TxInfo for direct Data conversion and usage "onchain"
-  Must keep all inputs and values ordered for determinism
+{- | Build draft tx body w/o setting ExBudget for inputs.
+ This body will be used to estimate `ExBudget` with `Cardano.Api` by `BodyBuilder`
 -}
-buildTxInfo ::
+buildDraftTx ::
   forall (w :: Type) (effs :: [Type -> Type]).
   Member (PABEffect w) effs =>
   PABConfig ->
+  Map PubKeyHash DummyPrivKey ->
   Tx ->
-  Eff effs (Either Text TxInfo)
-buildTxInfo pabConf tx = do
-  let txOutRefs = txInRef <$> Set.toList (txInputs tx) -- This will already be in order, for Sets listify acsending
-  eTxInInfos <- getTxInInfos @w txOutRefs
-  return $
-    (`second` eTxInInfos) $ \txInInfos ->
-      TxInfo
-        { txInfoInputs = txInInfos
-        , txInfoOutputs = txOutputs tx
-        , txInfoFee = sortValue $ txFee tx
-        , txInfoMint = sortValue $ txMint tx
-        , txInfoDCert = [] -- We don't support staking or stake redeeming at this time
-        , txInfoWdrl = []
-        , txInfoValidRange = slotRangeToPOSIXTimeRange (pcSlotConfig pabConf) $ txValidRange tx
-        , txInfoSignatories = sort $ Ledger.pubKeyHash <$> Map.keys (txSignatures tx)
-        , txInfoData = sortOn fst $ Map.toList $ txData tx
-        , txInfoId = Ledger.txId tx
-        }
-
--- | Sorts the internal maps
-sortValue :: Value -> Value
-sortValue (Value.Value m) = Value.Value $ sortMap $ sortMap PPrelude.<$> m
-  where
-    sortMap :: forall (k :: Type) (v :: Type). Ord k => AMap.Map k v -> AMap.Map k v
-    sortMap = AMap.fromList . sortOn fst . AMap.toList
+  Eff effs (Either Text FilePath)
+buildDraftTx pabConf privKeys tx = do
+  _ <- buildTx @w pabConf privKeys emptyBudget tx
```

Comment:
This caused a problem, when the cli failed with the build step, but the error message got thrown away. I fixed it in commit 96815667f7d510e321a820f9423e22094caa608c